### ISSUE TITLE
Safe handle has been closed crash

### ISSUE THREAD

**Issue by haf** [haf/expecto#239]
https://travis-ci.org/logary/logary/jobs/362324984#L1446

```
[20:32:28 INF] EXPECTO? Running tests... <Expecto>
Unhandled Exception: 
Unhandled Exception: System.ObjectDisposedException: Safe handle has been closed
   at System.Runtime.InteropServices.SafeHandle.DangerousAddRef(Boolean& success)
   at System.StubHelpers.StubHelpers.SafeHandleAddRef(SafeHandle pHandle, Boolean& success)
   at Microsoft.Win32.Win32Native.SetEvent(SafeWaitHandle handle)
   at System.Threading.EventWaitHandle.Set()
   at <StartupCode$FSharp-Core>.$Control.action@4236-16[TMsg](Mailbox`1 x, TMsg msg, Unit unitVar0)
   at Microsoft.FSharp.Control.Mailbox`1.Post(TMsg msg)
   at Expecto.Async.loop@261-6.Invoke(Unit unitVar)
   at Microsoft.FSharp.Control.AsyncBuilderImpl.econt@547.Invoke(ExceptionDispatchInfo exn)
--- End of stack trace from previous location where exception was thrown ---
   at System.Runtime.ExceptionServices.ExceptionDispatchInfo.Throw()
   at Microsoft.FSharp.Control.CancellationTokenOps.Start@958-1.Invoke(ExceptionDispatchInfo edi)
   at <StartupCode$FSharp-Core>.$Control.loop@124-50(Trampoline this, FSharpFunc`2 action)
   at Microsoft.FSharp.Control.Trampoline.ExecuteAction(FSharpFunc`2 firstAction)
   at Microsoft.FSharp.Control.AsyncImpl.ResultCell`1.RegisterResult(T res, Boolean reuseThread)
   at <StartupCode$FSharp-Core>.$Control.loop@124-50(Trampoline this, FSharpFunc`2 action)
   at Microsoft.FSharp.Control.Trampoline.ExecuteAction(FSharpFunc`2 firstAction)
   at <StartupCode$FSharp-Core>.$Control.-ctor@200-1.Invoke(Object o)
   at System.Threading.ExecutionContext.Run(ExecutionContext executionContext, ContextCallback callback, Object state)
   at System.Threading.ThreadPoolWorkQueue.Dispatch()
System.ObjectDisposedException: Safe handle has been closed
   at System.Runtime.InteropServices.SafeHandle.DangerousAddRef(Boolean& success)
   at System.StubHelpers.StubHelpers.SafeHandleAddRef(SafeHandle pHandle, Boolean& success)
   at Microsoft.Win32.Win32Native.SetEvent(SafeWaitHandle handle)
   at System.Threading.EventWaitHandle.Set()
   at <StartupCode$FSharp-Core>.$Control.action@4236-16[TMsg](Mailbox`1 x, TMsg msg, Unit unitVar0)
   at Microsoft.FSharp.Control.Mailbox`1.Post(TMsg msg)
   at Expecto.Async.loop@261-6.Invoke(Unit unitVar)
   at Microsoft.FSharp.Control.AsyncBuilderImpl.econt@547.Invoke(ExceptionDispatchInfo exn)
--- End of stack trace from previous location where exception was thrown ---
   at System.Runtime.ExceptionServices.ExceptionDispatchInfo.Throw()
   at Microsoft.FSharp.Control.CancellationTokenOps.Start@958-1.Invoke(ExceptionDispatchInfo edi)
   at <StartupCode$FSharp-Core>.$Control.loop@124-50(Trampoline this, FSharpFunc`2 action)
   at Microsoft.FSharp.Control.Trampoline.ExecuteAction(FSharpFunc`2 firstAction)
   at Microsoft.FSharp.Control.AsyncImpl.ResultCell`1.RegisterResult(T res, Boolean reuseThread)
   at <StartupCode$FSharp-Core>.$Control.loop@124-50(Trampoline this, FSharpFunc`2 action)
   at Microsoft.FSharp.Control.Trampoline.ExecuteAction(FSharpFunc`2 firstAction)
   at <StartupCode$FSharp-Core>.$Control.-ctor@200-1.Invoke(Object o)
   at System.Threading.ExecutionContext.Run(ExecutionContext executionContext, ContextCallback callback, Object state)
   at System.Threading.ThreadPoolWorkQueue.Dispatch()
```

**Comment by AnthonyLloyd** [comment]
I've rewritten these methods. I will release in the next day or so. Just trying to sort out the line endings changing on build.cmd.

**Comment by haf** [comment]
Another crash https://ci.appveyor.com/project/haf/logary/build/1.0.1062/job/ha1o4e8j60unuto9#L1055

**Comment by AnthonyLloyd** [comment]
Released in v7.0.1
